BEIMA.Backend/MongoService/MongoConnector.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MongoDB.Driver;
+using MongoDB.Bson;
+
+namespace BEIMA.Backend.MongoService
+{
+    public sealed class MongoConnector
+    {
+        //Public class members
+        public ServerType CurrentServerType { get; }
+
+        //Contains instance variables
+        private readonly MongoClient client = null;
+        private static readonly Lazy<MongoConnector> instance = new(() => new MongoConnector());
+
+        //Environment variables
+        private readonly string dbName = Environment.GetEnvironmentVariable("DatabaseName");
+        private readonly string deviceCollection = Environment.GetEnvironmentVariable("DeviceCollectionName");
+
+        //Singleton design pattern, used to get an instance of the MongoConnector
+        public static MongoConnector Instance { get { return instance.Value; } }
+
+        //Private constructor, used for singleton pattern. Cannot be called externally.
+        private MongoConnector()
+        {
+            string credentials;
+
+            if (Environment.GetEnvironmentVariable("CurrentEnv") == "dev-local")
+            {
+                CurrentServerType = ServerType.Local;
+                credentials = Environment.GetEnvironmentVariable("LocalMongoConnection");
+            }
+            else
+            {
+                CurrentServerType = ServerType.Cloud;
+                credentials = Environment.GetEnvironmentVariable("AzureCosmosConnection");
+            }
+            client = new MongoClient(credentials);
+        }
+
+        /*
+         * Checks if the client has been instantiated.
+         * Parameter: none
+         * Returns: true if client is not null, false if client is null.
+         */
+        public bool IsConnected()
+        {
+            return (client != null);
+        }
+
+        /*
+         * Inserts a device into the "devices" collection
+         * Parameter: BsonDocument that contains the fully formed device document (including all required and optional fields)
+         * Returns: ObjectId of the newly inserted object if successful, null if failed
+         */
+        public ObjectId? InsertDevice(BsonDocument doc)
+        {
+            if (!IsConnected())
```

Comment:
Since this check is so common, we could probably just put it in a method.

Reply:
We could probably even have a method to write the exception message and return null as well since that is used quite a bit too

Reply:
Fixed [f86d5e8](https://github.com/cs481-ekh/s22-beima/pull/67/commits/f86d5e8a2e85d6c7f43601eb50957275b48efcd8)